Updated the integration_test page (#5693)
flutter/website: src/docs/testing/integration-tests/index.md

**Hosts and targets**

During development, you are probably writing the code
on a desktop computer, called the **host** machine,
and running the app on a mobile device, browser,
or desktop application, called the **target** device.
(If you are using a web
browser or desktop application,
the host machine is also the target device.)

**flutter_driver** 

The `flutter_driver` package runs integration tests
written in Dart on a target device and reports the
results to the host. Tests written with `flutter_driver`
run from the host and drive the app running on a real
or virtual device. The `flutter drive` command is used
to run tests written with this package.

**integration_test**

Tests written with the `integration_test` package can:
 
1. Run directly on the target device, allowing you to test on
   multiple Android or iOS devices using Firebase Test Lab.
2. Run using `flutter_driver`. 
3. Use `flutter_test` APIs, making integration tests more
   like writing [widget tests][].

## Project setup

Add `integration_test`, `flutter_test`,
and optionally `flutter_driver` to your pubspec.yaml file:

```yaml
dev_dependencies:
  integraton_test:
    sdk: flutter

> flutter_test:
>   sdk: flutter
> flutter_driver:
>   sdk: flutter

Reviewer: Need to indent the other dependencies too:

```suggestion
  flutter_test:
    sdk: flutter
  flutter_driver:
    sdk: flutter
```

}
```

Instead of calling `driver.wait*` API,
use `WidgetTester.pumpAndSettle()` or
equivalent methods.

> [Android Device Testing]: {{site.pub-pkg}}/integration_test#android-device-testing

Reviewer: ```suggestion
[Android Device Testing]: {{site.github}}/flutter/flutter/tree/master/packages/integration_test#android-device-testing
```

[Download and install ChromeDriver]: https://chromedriver.chromium.org/downloads
[Firebase Console]: http://console.firebase.google.com/
[Firebase Test Lab]: https://firebase.google.com/docs/test-lab

> [Firebase Test Lab section of the README]: {{site.pub-pkg}}/integration_test#firebase-test-lab

Reviewer: ```suggestion
[Firebase Test Lab section of the README]: {{site.github}}/flutter/flutter/tree/master/packages/integration_test#firebase-test-lab
```

[flutter_test]: {{site.api}}/flutter/flutter_test/flutter_test-library.html
[`integration_test`]: {{site.github}}/flutter/flutter/tree/master/packages/integration_test#integration_test

> [integration_test usage]: {{site.pub-pkg}}/integration_test#usage

Reviewer: ```suggestion
[integration_test usage]: {{site.github}}/flutter/flutter/tree/master/packages/integration_test#usage
```

> [iOS Device Testing]: {{site.pub-pkg}}/integration_test#ios-device-testing

Reviewer: ```suggestion
[iOS Device Testing]: {{site.github}}/flutter/flutter/tree/master/packages/integration_test#ios-device-testing
```
